src/contracts/core/AVSDirectory.sol

```diff
@@ -67,24 +67,63 @@ contract AVSDirectory is
     function createOperatorSets(uint32[] calldata operatorSetIds) external {
         for (uint256 i = 0; i < operatorSetIds.length; ++i) {
             require(
-                !isOperatorSet[msg.sender][operatorSetIds[i]], "AVSDirectory.createOperatorSet: operator set already exists"
+                !isOperatorSet[msg.sender][operatorSetIds[i]],
+                "AVSDirectory.createOperatorSet: operator set already exists"
             );
             isOperatorSet[msg.sender][operatorSetIds[i]] = true;
-            emit OperatorSetCreated(msg.sender, operatorSetIds[i]);        
+            emit OperatorSetCreated(msg.sender, operatorSetIds[i]);
         }
     }
 
     /**
      * @notice Sets the AVS as an operator set AVS, preventing legacy M2 operator registrations.
-     * 
-     * @dev msg.sender must be the AVS. 
+     *
+     * @dev msg.sender must be the AVS.
      */
     function becomeOperatorSetAVS() external {
         require(!isOperatorSetAVS[msg.sender], "AVSDirectory.becomeOperatorSetAVS: already an operator set AVS");
         isOperatorSetAVS[msg.sender] = true;
         emit AVSMigratedToOperatorSets(msg.sender);
     }
 
+    /**
+     * @notice Called by an AVS to migrate operators that have a legacy M2 registration to operator sets.
+     *
+     * @param operators The list of operators to migrate
+     * @param operatorSetIds The list of operatorSets to migrate the operators to
+     *
+     * @dev The msg.sender used is the AVS
+     * @dev The operator can only be migrated at most once per AVS
+     * @dev The AVS can no longer register operators via the legacy M2 registration path once it begins migration
+     * @dev The operator is deregistered from the M2 legacy AVS once migrated
+     */
+    function migrateOperatorsToOperatorSets(
+        address[] calldata operators,
+        uint32[][] calldata operatorSetIds
+    ) external override onlyWhenNotPaused(PAUSED_OPERATOR_REGISTER_DEREGISTER_TO_AVS) {
+        // Assert that the AVS is an operator set AVS.
+        require(isOperatorSetAVS[msg.sender], "AVSDirectory.migrateOperatorsToOperatorSets: AVS is not an operator set AVS");
+
+        for (uint256 i = 0; i < operators.length; i++) {
```

Comment:
Don't we need another for loop to register for each operator set for each operator?